backend/src/test/java/com/redhat/cloud/notifications/routers/NotificationResourceTest.java

```diff
@@ -330,6 +331,114 @@ void testEventTypeFetchingByBundleApplicationAndEventTypeName() {
+        List<Application> apps = applicationRepository.getApplications(TEST_BUNDLE_NAME);
+        Application app1 = apps.stream().filter(a -> a.getName().equals(TEST_APP_NAME)).findFirst().get();
+        UUID appId1 = app1.getId();
+        Application app2 = apps.stream().filter(a -> a.getName().equals(TEST_APP_NAME_2)).findFirst().get();
+        UUID appId2 = app2.getId();
+        Header identityHeader = initRbacMock(ACCOUNT_ID, ORG_ID, "user", FULL_ACCESS);
+        UUID behaviorGroupId1 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-1", bundleId).getId();
+        UUID behaviorGroupId2 = helpers.createBehaviorGroup(ACCOUNT_ID, ORG_ID, "behaviour-group-2", bundleId).getId();
+
+        // bgroup1 assigned to ev0 and ev1 on TEST_APP_NAME, bgroup2 assigned to ev1 on TEST_APP_NAME_2, all other event types unassigned
+        UUID endpointId1 = helpers.createWebhookEndpoint(ACCOUNT_ID, ORG_ID);
+        List<EventType> eventTypesApp1 = applicationRepository.getEventTypes(appId1);
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypesApp1.getFirst().getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypesApp1.get(1).getId(), Set.of(behaviorGroupId1));
+        behaviorGroupRepository.updateBehaviorGroupActions(ORG_ID, behaviorGroupId1, List.of(endpointId1));
+        UUID endpointId2 = helpers.createWebhookEndpoint(ACCOUNT_ID, ORG_ID);
+        List<EventType> eventTypesApp2 = applicationRepository.getEventTypes(appId2);
+        behaviorGroupRepository.updateEventTypeBehaviors(ORG_ID, eventTypesApp2.get(1).getId(), Set.of(behaviorGroupId2));
+        behaviorGroupRepository.updateBehaviorGroupActions(ORG_ID, behaviorGroupId2, List.of(endpointId2));
+
+        Response unmutedResponse = given()
+                .when()
+                .header(identityHeader)
+                .queryParam("bundleId", bundleId)
+                .queryParam("applicationIds", appId1)
+                .queryParam("eventTypeName", "1")
+                .queryParam("excludeMutedTypes", "true")
+                .get("/notifications/eventTypes")
+                .then()
+                .statusCode(200)
+                .contentType(JSON)
+                .extract().response();
+
+        JsonObject unmutedPage = new JsonObject(unmutedResponse.getBody().asString());
+        JsonArray unmutedEventTypes = unmutedPage.getJsonArray("data");
+        for (int i = 0; i < unmutedEventTypes.size(); i++) {
+            JsonObject unmutedEv = unmutedEventTypes.getJsonObject(i);
+            unmutedEv.mapTo(EventType.class);
+            assertEquals(bundleId.toString(), unmutedEv.getJsonObject("application").getString("bundle_id"));
+            assertEquals(appId1.toString(), unmutedEv.getJsonObject("application").getString("id"));
+            assertTrue(unmutedEv.getString("display_name").contains("1") || unmutedEv.getString("name").contains("1"));
+        }
```

Comment:
Could you please add an assertion to verify that the response only contains one event type?